components/atomic/dropdown/Dropdown/BaseDropdown.tsx

```diff
@@ -1,75 +1,81 @@
-import React, { useState, useRef, useCallback } from "react";
+import React, { useState, useRef, useCallback, forwardRef } from "react";
 import { useUID } from "react-uid";
 import { useFocusTrap } from "@castiron/hooks/";
 import isFunction from "lodash/isFunction";
+import * as Styled from "./Dropdown.styles";
 
 /**
  * A dropdown for navigation submenus.
  * Adds appropriate aria labels and open/close functionality.
  */
-function BaseDropdown({ className, disclosure, submenu, label }: Props) {
-  const uid = useUID();
-  const wrapperRef = useRef(null);
-  const [active, setActive] = useState(false);
+const BaseDropdown = forwardRef<HTMLUListElement, Props>(
+  ({ className, disclosure, menuItems, out, label }, elRef) => {
+    const uid = useUID();
+    const wrapperRef = useRef(null);
+    const [active, setActive] = useState(false);
 
-  useFocusTrap(wrapperRef, active, {
-    onDeactivate: useCallback(() => setActive(false), [setActive]),
-  });
+    const closeDropdown = useCallback(() => setActive(false), [setActive]);
+    useFocusTrap(wrapperRef, active, {
+      onDeactivate: closeDropdown,
+    });
 
-  const disclosureProps = {
-    "aria-controls": uid,
-    "aria-expanded": active,
-    "aria-label": `Show ${label}`,
-    onClick: () => setActive((prevState) => !prevState),
-  };
+    const submenu = (
+      <Styled.List
+        ref={elRef}
+        right={out.right}
+        id={uid}
+        aria-hidden={!active}
+        hidden={!active}
+      >
+        {menuItems.map((item, i) => {
+          if (item === null) return null;
+          return (
+            <Styled.Item key={i}>
+              {React.cloneElement(item, { closeDropdown: closeDropdown })}
+            </Styled.Item>
+          );
+        })}
+      </Styled.List>
+    );
```

Comment:
Here's the new `submenu` with the `forwardRef` and a lot less cloning. I think it's a lot easier to follow now. Since we're passing `closeDropdown` to a `ButtonControlGroup` here, each button in the list can get that prop and thus close the dropdown.

A question, though— now that `submenu` is here, styles are too. So, if we go this route, it's not clear to me whether it's still useful to have two components, `Dropdown` and `BaseDropdown`. Potentially, we could also pull `submenu` out as its own component?